libs/logic-apps-shared/src/utils/src/lib/helpers/functions.ts

```diff
@@ -856,7 +856,7 @@ export function deepCompareObjects(object1: Record<string, any> | undefined, obj
     const val2 = object2[key];
     const areObjects = isObject(val1) && isObject(val2);
 
-    if ((areObjects && !deepCompareObjects(val1, val2)) || (!areObjects && val1 !== val2)) {
+    if (areObjects ? !deepCompareObjects(val1, val2) : JSON.stringify(val1) !== JSON.stringify(val2)) {
```

Comment:
Just making sure - Do we care about case-insensitive behavior here?

Reply:
We should be case-sensitive here, so just the `!==` should be fine